### PR TITLE
Schema

### PR DESCRIPTION
**Pull request by Xfarya**
added schema with a test to check if all fields are entered

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,8 +23,10 @@
         "typescript": "^4.6.3"
     },
     "dependencies": {
+        "chai": "^4.3.6",
```
Comment (edpackard-work) [inline package.json]: Place chai and sinon in devDependencies - or use Jest for testing (see init Backend ticket)

File: src/model/Product.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
```
Comment (Si3rr4wow) [inline src/model/Product.ts]:
```suggestion
import { Schema } from 'mongoose'
```

File: package.json [hunk 3]
```diff
@@ -23,8 +23,10 @@
...
         "dotenv": "^16.0.0",
         "express": "^4.17.3",
-        "mongoose": "^6.2.9"
+        "mongoose": "^6.2.9",
+        "sinon": "^13.0.1"
```
Comment (Si3rr4wow) [inline package.json]: drop sinon and chai in favour of jest

File: src/__test__/productSchema.spec.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+var Product = require('../model/Product');
+
+test("Should be invalid if all fields are empty", done => {
+    const product = new Product(); 
+    product.validate((err: any) => {
+      expect(err).not.toBeNull();
+      done();
+    });
+  });
```
Comment (Si3rr4wow) [inline src/__test__/productSchema.spec.ts]: we can drop this test

**Pull request merged**